File: practica_final/GameShop-1/GameShop/app/src/main/java/com/example/t11navigation/fragments/FragmentCarrito.java
package com.example.t11navigation.fragments;


import static com.example.t11navigation.fragments.FragmentCompras.carrito;

import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.t11navigation.ModelJuego;
import com.example.t11navigation.R;
import com.example.t11navigation.adapters.AdaptadorCarrito;
import com.example.t11navigation.adapters.AdaptadorJuego;

import java.net.URISyntaxException;
import java.util.ArrayList;

public class FragmentCarrito extends Fragment {

    RecyclerView recyclerView;

    private View view;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        view = inflater.inflate(R.layout.fragment_carrito, container, false);

        TextView total = view.findViewById(R.id.txt_total);

        recyclerView = view.findViewById(R.id.recviewcarrito);

        if(carrito.size() > 0) {

        recyclerView.setLayoutManager(new GridLayoutManager(getContext(),1));
        recyclerView.setAdapter(new AdaptadorCarrito(carrito));


            double T = 0.0;

            for (int i = 0; i < carrito.size(); i++) {
                T += Integer.parseInt(carrito.get(i).getPrecioJuego().substring(0,carrito.get(i).getPrecioJuego().length()-1));
            }

            total.setText("" + T);
        }else
        {
            total.setText("0");
        }

        return view;
    }
}

File: practica_final/practica_final/app/src/main/java/com/example/practica_final/FragmentCarrito.kt
package com.example.practica_final

import android.content.Context
import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.LinearLayoutManager
import com.bumptech.glide.Glide
import com.example.practica_final.adapter.AdaptadorCarrito
import com.example.practica_final.adapter.AdaptadorProductos
import com.example.practica_final.databinding.FragmentCarritoBinding
import com.example.practica_final.databinding.FragmentDetalleBinding
import com.example.practica_final.model.Producto


class FragmentCarrito : Fragment() {
    private lateinit var secondActivity: SecondActivity
    private lateinit var binding: FragmentCarritoBinding
    private lateinit var adaptadorCarrito: AdaptadorCarrito


    override fun onAttach(context: Context) {
        super.onAttach(context)

        secondActivity = requireActivity() as SecondActivity
        var precio: Int = 0
        for (producto in secondActivity.productosCarrito){
            precio += producto.precio
        }

        adaptadorCarrito = AdaptadorCarrito(secondActivity.productosCarrito, context)
    }

    public fun modificarPrecioTotal(value: Int){
        var precio = binding.textoPrecioTotal.text.toString().toInt()
        precio += value

        binding.textoPrecioTotal.text = precio.toString()
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding = FragmentCarritoBinding.inflate(inflater,container,false)

        binding.recyclerCarrito.adapter = adaptadorCarrito
        binding.recyclerCarrito.layoutManager = LinearLayoutManager(context,
            LinearLayoutManager.VERTICAL,false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        binding.botonFinalizarCompra.setOnClickListener {
            // Cambiamos a 0 el precio total de nuevo y cambiamos al fragment de productos
            binding.textoPrecioTotal.text = "0"
            secondActivity.productosCarrito = ArrayList()

            parentFragmentManager.popBackStack()
        }

    }

    override fun onResume() {
        super.onResume()

        var precio = binding.textoPrecioTotal.text.toString().toInt()

        for (producto in secondActivity.productosCarrito){
            precio += producto.precio
        }

        binding.textoPrecioTotal.text = precio.toString()
    }
}